plugins/modules/dynamic_group.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2024, Network to Code (@networktocode) <info@networktocode.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: dynamic_group
+short_description: Creates or removes dynamic groups from Nautobot
+description:
+  - Creates or removes dynamic groups from Nautobot
+notes:
+  - Tags should be defined as a YAML list
+  - This should be ran with connection C(local) and hosts C(localhost)
+author:
+  - Travis Smith (@tsm1th)
+requirements:
+  - pynautobot
+version_added: "5.5.0"
+extends_documentation_fragment:
+  - networktocode.nautobot.fragments.base
+  - networktocode.nautobot.fragments.tags
+  - networktocode.nautobot.fragments.custom_fields
+options:
+  name:
+    description:
+      - The name of the dynamic group
+    required: true
+    type: str
+  description:
+    description:
+      - The description of the dynamic group
+    required: false
+    type: str
+  group_type:
+    description:
+      - Required if I(state=present) and the dynamic group does not exist yet
+      - V(static) is only a valid choice in Nautobot 2.3+
+    choices: [ dynamic-filter, dynamic-set, static ]
+    required: false
+    type: str
```

Comment:
I ran the [full test suite](https://github.com/nautobot/nautobot-ansible/actions/runs/12144671555) on your branch and It looks like the `group-type` field isn't available pre-2.3 (not just the static choice), so we will need to omit it completely for earlier versions:

```
integration-1  | fatal: [testhost]: FAILED! => {"msg": "The conditional check 'test_create_min['dynamic_group']['group_type'] == \"dynamic-filter\"' failed. The error was: error while evaluating conditional (test_create_min['dynamic_group']['group_type'] == \"dynamic-filter\"): 'dict object' has no attribute 'group_type'. 'dict object' has no attribute 'group_type'"}
```

Reply:
There are a cascade of issues trying to support pre-2.3 in the tests.  I fixed the `group_type`, but the next thing I hit is that `filter` is required on all groups created pre-2.3 which is not the case for post-2.3.  

Is this really worth it?

Reply:
If so, then it might be easier to just do two sets of tests at this point.

Reply:
2 sets of tasks are fine, I just didn't want to gate the entire module tests.